### PR TITLE
Futher improvements and refinements to the (environment) documentation

### PR DESCRIPTION
**Pull request by Darwinkel**
### Changes
Adds various (misc) changes to the (environment) docs, including:

- Using `DirectoryPath` and `FilePath` instead of generic `Path` in Pydantic's validator
  - Keiko now explicitly requires creating the `/reports` folder beforehand 
- Removing and updating old documentation

### Issue link
Relates to https://github.com/minvws/nl-kat-coordination/issues/1580. Resolves a majority of the sub tasks in that ticket.

### Proof
See CI and CI artifacts.

---

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified;
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have made corresponding changes to the documentation, if necessary.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Review by praseodym — APPROVED** [review]
Nice update, also thanks for removing old information from the docs!

File: .github/workflows/keiko_test.yml [hunk 1]
```diff
@@ -44,6 +44,11 @@ jobs:
...
         working-directory: ./keiko
 
+      - name: Create writeable directory for reports (edge case for GitHub Actions)
+        run: mkdir -p /tmp/keiko/reports && chmod -R 700 /tmp/keiko/reports
+
       - name: Run unit tests
         run: make test
         working-directory: ./keiko
+        env:
+          KEIKO_REPORTS_FOLDER: /tmp/keiko/reports
```
Comment (dekkers) [inline .github/workflows/keiko_test.yml:54]: Missing newline at the and of the file. pre-commit didn't fix this because of #1660

Comment (underdarknl) [reply .github/workflows/keiko_test.yml:54]: 1660 is now merged.

**Review by ammar92 — APPROVED** [review]

**Review by dekkers — APPROVED** [review]
